tsfc/kernel_interface/common.py

```diff
@@ -58,6 +58,11 @@ def cell_orientation(self, restriction):
                                                gem.Literal(1),
                                                gem.Literal(numpy.nan)))
 
+    def cell_size(self, restriction):
+        f = {None: (), '+': (0, ), '-': (1, )}[restriction]
+        # cell_sizes expression must have been set up by now.
+        return gem.partial_indexed(self._cell_sizes, f)
+
```

Comment:
`cell_size` is like `cell_orientations` that they are calculated once globally?

Reply:
Indeed.